mainwindow.h

```diff
@@ -36,8 +36,8 @@ private slots:
     void on_localModels_activated(int index);
 
 private:
-    Ui::MainWindow *ui;
-    MarianInterface * translator_; // Initialised to nullptr. Can't use smart ptrs due to destructor needing to run
+    std::unique_ptr<Ui::MainWindow> ui_;
+    std::unique_ptr<MarianInterface> translator_;
```

Comment:
I disabled this, because we can't use `.reset()`, as it first creates a new object, later destroys old one and one could easily crash the application if they don't know that you can't use this method.